NumberBaseball/NumberBaseball/main.swift

```diff
@@ -35,19 +32,32 @@ func evaluateStrikeBall(computer computerNumbers: [Int], player playerNumbers: [
     return (strike, ball)
 }
 
-func playNumberBaseballGame(remainingChance: inout Int, computerNumbers: inout [Int]) {
+func playNumberBaseballGame() {
+    let computerNumbers = generateNumbers()
+    var playerNumbers: [Int] = []
+    var remainingChance = 9
+    
     while remainingChance > 0 {
-        remainingChance -= 1
-        
-        var ball = 0
         var strike = 0
+        var ball = 0
+        
+        print("숫자 3개를 띄어쓰기로 구분하여 입력해주세요.\n중복 숫자는 허용하지 않습니다.\n입력 : ", terminator: "")
+        let input = readLine() ?? ""
         
-        computerNumbers = generateNumbers()
-        let playerNumbers = generateNumbers()
+        playerNumbers = input
+            .split(separator: " ")
+            .compactMap { Int($0 ) }
+            .filter { $0 > 0 && $0 < 10 }
+        
```

Comment:
고차함수를 연결해서 사용하셨네요 👍

`compactMap`과 `map`은 어떤 타입에서 사용할 수 있을까요?
둘의 차이점은 무엇일까요?

Reply:
```suggestion
        playerNumbers = input
            .split(separator: " ")
            .compactMap { Int($0) }
            .filter { $0 > 0 && $0 < 10 }
        
```
Int로 형 변환 하는 부분에 공백이 하나 더 추가되어 있네요. 이 부분은 수정이 필요할 것 같습니다!

Reply:
`compactMap`과 `map`은 컬렉션 타입에서 사용할 수 있습니다.
`compactMap`은 `nil`을 포함하지 않습니다.

Reply:
compactMap과 map은 sequence 프로토콜을 준수하는 모든 타입에서 사용할 수 있습니다.